### DISCUSSION TITLE
Strange issue Syslog-ng 3.24.1 and libssh or libssh2

### DISCUSSION THREAD

**Discussion by karlstiller** [syslog-ng/syslog-ng#4404] [Q&A] [answered]
I'm facing a strange issue regarding the use of libssh with syslog-ng 3.24.1.

I'm running on an embedded Linux target board and have been using syslog-ng for some time now. Recently there was a change in the list of packages contained in the image, the package libssh2 was swapped out for the libssh package. The unrelated change was made in order to allow sftp and curl on the target. This change caused an issue in syslog-ng when attempting to open /proc/kmsg as a source. The exact configuration setting snippet which caused the issue is below:
`source s_kern { file("/proc/kmsg" program_override("kernel"));
};`

The error message I receive is:
`[2023-03-22T14:54:57.396674] Error opening file for reading; filename='/proc/kmsg', error='Operation not permitted (1)'
[2023-03-22T14:54:57.398177] Error initializing message pipeline; plugin_name='file', location='/etc/syslog-ng/syslog-ng.conf:26:17'`

If I swap back out libssh for libssh2 the problem is solved for syslog-ng, but breaks sftp.

Does syslog-ng depend on libssh2 indirectly somehow? Or perhaps is there a way around this issue?

Thanks

Karl

**Comment by bazsi** [comment, marked as answer]
Hi,

No it does not use libssh/libssh2

Maybe something else opens /proc/kmsg if those packages are installed? It's
the kernel that refuses to open that file for some reason.

You might also want to look at dmesg for clues wether selinux/apparmor/etc
denies this request for a reason.

**Reply by karlstiller** [reply]
I had a look in the kernel logs, but there were no messages for any reason for the failure. However, we did find that syslog-ng was linking by default with libcurl for http. As http isn't needed for our situation, disabling http allowed syslog-ng to operate as normal. http was disabled by adding the following line in the yocto/bitbake local.conf configuration script:
`PACKAGECONFIG_remove_pn-syslog-ng=" http"`

**Comment by bazsi** [comment]
I don't see how that could relate to the failure of opening /proc/kmsg
